engines/wintermute/base/saveload.cpp

```diff
@@ -159,31 +159,39 @@ void SaveLoad::afterLoadScript(void *script, void *data) {
 }
 
 Common::String SaveLoad::getSaveSlotFilename(int slot) {
+	Common::String filename;
 	BasePersistenceManager *pm = new BasePersistenceManager();
-	Common::String filename = pm->getFilenameForSlot(slot);
-	delete pm;
+	if (pm) {
```

Comment:
When can this fail (silently)?

Reply:
Hmmmm... No idea, actually.
I guess I was distracted by `if (!pm) { return false;}` in original code of `getSaveSlotDescription`.

Can we leave it as is?
Should I refactor it again & remove all `if (pm)` checks?
Should I refactor it again & add more `if (pm)` checks? =)

Reply:
This particular check makes no sense to keep, so I'd suggest to remove it. Not sure about the other checks you're referring to

Reply:
Removed at 48ea8b6.